### DISCUSSION TITLE
Client flavours - a dumb idea?

### DISCUSSION THREAD

**Discussion by natdab** [serverpod/serverpod#2623] [General]
Most using Serverpod I would expect generally are building one app/client to interact with their server, and in almost all cases I would imagine this is absolutely the right path to go down. It may simply be my (admittedly poorly formed) use case, but for those that may have multiple apps in which only certain server functionality would be exposed, being able to specify a "flavour" for a client in models (and endpoints is possible) could be helpful.

This would essentially scope client code generation to a specific flavour of client, and not include models or endpoints that do not have this flavour noted. Flavoured clients would exist in a subfolder ie _mypod_client_flavour_. Created a flavoured client currently involves removing significant generated code manually.

I recognise that in most cases this is likely to be quite dumb - especially if only building for mobile apps, but if building for the web (despite obfuscation) it isn't a terrible idea to limit what is generated if it can be limited.

Got the end and noticed I'd used only UK English spelling! Apologies to those elsewhere. Flavour == flavor (added for those that may search for similar).

**Comment by Isakdl** [comment]
If I understand your use case, you have like a consumer app and then an admin app and would like to split the code shipped to each client library individually. 

It is an interesting idea, but I wonder what your needs are. Do you want to avoid shipping "admin" code with the consumer app or do you want also to hide the endpoints from developers? 

If the former is good enough, then having a shared library in practice is okay as Dart will tree-shake the code at compile time. This means no unused methods will be included in the binary. Hence if an admin endpoint is not in use in the consumer app the final app will not have the code to call the admin endpoint.
